src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets

```diff
@@ -310,6 +312,7 @@ The .NET Foundation licenses this file to you under the MIT license.
       <CustomLinkerArg Include="/LIBPATH:&quot;%(AdditionalNativeLibraryDirectories.Identity)&quot;" Condition="'$(_targetOS)' == 'win' and '@(AdditionalNativeLibraryDirectories->Count())' &gt; 0" />
       <CustomLinkerArg Include="-exported_symbols_list &quot;$(ExportsFile)&quot;" Condition="'$(_IsApplePlatform)' == 'true' and $(ExportsFile) != ''" />
       <CustomLinkerArg Include="-Wl,--version-script=$(ExportsFile)" Condition="'$(_targetOS)' != 'win' and '$(_IsApplePlatform)' != 'true' and $(ExportsFile) != ''" />
+      <CustomLinkerArg Include="-Wl,--export-dynamic" Condition="'$(_targetOS)' != 'win' and '$(_IsApplePlatform)' != 'true' and $(ExportsFile) != '' and '$(NativeLib)' == ''" />
```

Comment:
This change is linux/ELF format specific and is required as on these platforms exported symbols of an executable are not placed in the dynamic symbol table. This prevents `dlopen` to find the desired symbols. It is also worth noting that this should not be confused with `--version-script` which only makes sure that the symbols are exported - made global.
`--export-dynamic` is documented in: https://linux.die.net/man/1/ld  

---
Here is a dynamic symbol table dump with and without the linker switch for the `GenerateUnmanagedEntryPoints` test added in this PR 

`readelf --dyn-syms <program>`
| Without '--export-dynamic'  | With '--export-dynamic' |
| ------------- | ------------- |
|Symbol table '.dynsym' contains 153 entries: | Symbol table '.dynsym' contains 155 entries: |
|   Num:    Value          Size Type    Bind   Vis      Ndx Name |    Num:    Value          Size Type    Bind   Vis      Ndx Name |
|     0: 0000000000000000     0 NOTYPE  LOCAL  DEFAULT  UND |     0: 0000000000000000     0 NOTYPE  LOCAL  DEFAULT  UND |
| ... | ... |
|   152: 0000000000000000     0 OBJECT  GLOBAL DEFAULT  ABS V1.0 |    152: 00000000000d0230     0 NOTYPE  GLOBAL DEFAULT   16 ReferencedAssembly1Method@@V1.0 |
| |   153: 0000000000000000     0 OBJECT  GLOBAL DEFAULT  ABS V1.0 |
| |   154: 00000000002b3cf0     0 NOTYPE  GLOBAL DEFAULT   16 MainAssemblyMethod@@V1.0 |

---

Reply:
Looks good to me. Thank you!